src/java.base/share/classes/jdk/internal/module/ModuleBootstrap.java

```diff
@@ -471,13 +480,13 @@ private static ModuleLayer boot2() {
 
         // Archive module graph and boot layer can be archived at CDS dump time.
         // Only allow the unnamed module case for now.
-        if (canArchive && (mainModule == null)) {
+        if (canArchive) {
```

Comment:
Doesn't this invalidate the comments?

Reply:
I'll update the comment in my next update.
I've just pushed a commit based on some offline discussion with Ioi.